### PR TITLE
arch & build/configs: Enable CONFIG_ARCH_VECNOTIRQ in all esp32 confi…

### PR DESCRIPTION
**Pull request by xiarihf**
…guration file.

    1. Enable CONFIG_ARCH_VECNOTIRQ in all esp32 configuration file.
    2. Select ARCH_VECNOTIRQ when Xtensa architecture is selected in menuconfig.

	CONFIG_ARCH_VECNOTIRQ is added for the cases that the irq vectors are not cpu IRQs.
	Such as there is a 1-to-many relationship between vector numbers and interrupt numbers or different.
    Xtensa has more then 32 interrupts, the vector indexes and interrupt numbers are different.
    So we need to enable CONFIG_ARCH_VECNOTIRQ to avoid irq_attch function to do the wrong work.

Signed-off-by: xiarihf qiang3.zhang@samsung.com

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e92071e657c57e0666517b927c8c3617579598d6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e92071e657c57e0666517b927c8c3617579598d6] - Code Rule Check (C++) OK.

File: os/arch/Kconfig [hunk 1]
```diff
@@ -21,6 +21,7 @@ config ARCH_XTENSA
 	bool "Xtensa"
 	select ARCH_HAVE_STACKCHECK
 	select ARCH_HAVE_CUSTOMOPT
+	select ARCH_VECNOTIRQ
 	---help---
 		Cadence Tensilica Xtensa architectures.
 endchoice
```
Comment (ChinaTizenRT) [inline os/arch/Kconfig]: why omit esp32_DevKitC/hello_with_tash to add CONFIG_ARCH_VECNOTIRQ
why not add it on esp32_DevKitC board.

**Comment by seinfra** [comment]
Target : [71c318d68a510c8deeb9dd3a1afeb35b99052d54] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [71c318d68a510c8deeb9dd3a1afeb35b99052d54] - Code Rule Check (C++) OK.

**Comment by xiarihf** [comment]
@sunghan-chang 

Hi, Mr. chang, can you help to review this commit?
Please help to merge it, if it's OK, Thanks.

File: os/arch/Kconfig (hunk 1 above)
Comment (sunghan-chang) [reply os/arch/Kconfig]: There are 7 defconfigs for esp32 board. Why do you add it on only 3?

Comment (xiarihf) [reply os/arch/Kconfig]: Others defconfig files are already added.  
The new submitted is not included by the submitter.

**Pull request merged**